log/js/dashboard.js

```diff
@@ -350,13 +352,24 @@ define([
  					this.errorRatio[userIndex][problemIndex] = errorRatioText;
 					
 					this.problemComplete[userIndex][problemIndex] = upObject['problemComplete'];
+					var nodesCorrectRatioText = "Blank";
 					if(this.problemComplete[userIndex][problemIndex] == true){
 						this.totalProblemsCompleted[userIndex]++;
 						this.totalProblemsStarted[userIndex]++;
+						var nodesCorrect = 0;
+						array.forEach(upObject["nodes"], function(node){
+							if(node.isNodeCorrect){
```

Comment:
Add this code to php, this processing is part of that. Only final display changes are what I used to do in JS. The idea being we are already going over each node in PHP while going through logs, then why should we add it again here. It is already very heavy because it has to go through each student each problem and so on.

Reply:
And while you are at it would it be possible to fix the indentation of whole file. Just open it in sublime and set 4 spaces as a tab. If possible push it as a separate commit.